src/gluonnlp/model/bert.py

```diff
@@ -385,18 +392,23 @@ def forward(self, inputs, token_types, valid_length=None, masked_positions=None)
         This is used in training or fine-tuning a BERT model.
         """
         outputs = []
-        seq_out, _ = self._encode_sequence(inputs, token_types, valid_length)
+        seq_out, attention_out = self._encode_sequence(inputs, token_types, valid_length)
         outputs.append(seq_out)
+        output = seq_out[-1] if isinstance(seq_out, list) else seq_out
```

Comment:
Could you add some assertion?
```
if self._output_all_encodings:
    assert isinstance(seq_out, list)
    output = seq_out[-1]
else:
    output = seq_out
```

Reply:
Added, though `_output_all_encodings` is a private field of encoder, so technically it breaks encapsulation here. But I guess it is fine.